roles/etcd/tasks/gen_nodes_certs_script.yml

```diff
@@ -14,6 +14,8 @@
     - "{{ my_etcd_node_certs }}"
 
 - name: Gen_certs | Gather node certs
+  vars:
+    ansible_ssh_retries: 10
```

Comment:
Not sure about setting a ansible_ssh_retries only here. I think if there is something to change it should probably be  done globally, possibly in many tasks or even maybe having a special dummy task early to "warmup" the ssh connection (as they should be kept aftewards).

Reply:
Well, I only put it here because from a quick grep, it looks like this is the only task with the following:
- `delegate_to`
- no `run_once`
- acts on all nodes (or at least most of them, not only etcd/control plane)

Which produces the high numbers of ssh connections.
I'm not sure a "warmup" task would do anything, because this is more about sessions than connections really. Basically MaxSessions in sshd applies to multiplexing ("Specifies  the  maximum  number of open shell, login or subsystem (e.g. sftp) sessions permitted per network connection").

Doing that change globally would not be a good idea, I think, because it would prevent "failing fast" on the whole playbook.

Reply:
... and this makes me wonder if that problems with MaxSessions would also happens if that tasks was not using `shell`. I'm not sure how that's working precisely in ansible, delegate_to task might be coalesced somehow when not using the shell module... :thinking:

Reply:
Ah ok I see, then all good on my side for this patch. However Increasing MaxSession might be something that we could do in Kubespray if you are operating on a high enough number of nodes somehow (we could do that only on the first etcd nodes + first control plane nodes?).

Reply:
Well, yeah, we could. That would require  to restart sshd though, and I found it intrusive than necessary for that one task.

Reply:
Iirc it's not super intrusive to do that, for instance it somehow preserve the current connection and you don't need to reconnect and so on

Reply:
(that's probably a subject for another patch in any case)